### PR TITLE
Add code linting on pull request

### PR DESCRIPTION
**Pull request by Ninja243**
Lints code using eslint using the airbnb standard when a pull request is made.

### CODE REVIEW

**Review by aavshr — COMMENTED** [review]
Thanks @Ninja243. The `eslint` config requires rules and consultation from @xeust based on our current javascript linting rules used. I'd suggest to keep this PR open for now.

**Comment by Ninja243** [comment]
Sounds good me, thanks for taking a look! 👍

Aavash Shrestha ***@***.***> schrieb am Do., 8. Apr. 2021,
10:46:

**Comment by xeust** [comment]
Hi @aavshr, @Ninja243,

Here is the `.eslintrc.js` file from the frontend. There is a lot of non-relevant `react` and `ES6` stuff in it, but we use both the `airbnb` eslint standard alongside `prettier`. I can also create a gist out of it, just let me know!


`module.exports = {
    "extends": [
      "airbnb",
      "prettier",
      "prettier/react"
    ],
    "parser": "babel-eslint",
    "parserOptions": {
      "ecmaVersion": 2018,
      "ecmaFeatures": {
        "impliedStrict": true,
        "classes": true
      }
    },
    "env": {
      "browser": true,
      "node": true,
      "jquery": true,
      "jest": true
    },
    "rules": {
      "no-debugger": 0,
      "no-alert": 0,
      "no-await-in-loop": 0,
      "no-return-assign": [
        "error",
        "except-parens"
      ],
      "no-restricted-syntax": [
        2,
        "ForInStatement",
        "LabeledStatement",
        "WithStatement"
      ],
      "no-unused-vars": [
        1,
        {
          "ignoreSiblings": true,
          "argsIgnorePattern": "res|next|^err"
        }
      ],
      "prefer-const": [
        "error",
        {
          "destructuring": "all",
        }
      ],
      "arrow-body-style": [
        2,
        "as-needed"
      ],
      "no-unused-expressions": [
        2,
        {
          "allowTaggedTemplates": true
        }
      ],
      "no-param-reassign": [
        2,
        {
          "props": false
        }
      ],
      "no-console": 0,
      "import/prefer-default-export": 0,
      "import": 0,
      "func-names": 0,
      "space-before-function-paren": 0,
      "comma-dangle": 0,
      "max-len": 0,
      "import/extensions": 0,
      "no-underscore-dangle": 0,
      "consistent-return": 0,
      "react/display-name": 1,
      "react/no-array-index-key": 0,
      "react/react-in-jsx-scope": 0,
      "react/prefer-stateless-function": 0,
      "react/forbid-prop-types": 0,
      "react/no-unescaped-entities": 0,
      "jsx-a11y/accessible-emoji": 0,
      "react/require-default-props": 0,
      "react/prop-types": 0,
      "import/named": 0,
      "jsx-a11y/no-autofocus":0,
      "react/jsx-filename-extension": [
        1,
        {
          "extensions": [
            ".js",
            ".jsx"
          ]
        }
      ],
      "radix": 0,
      "no-shadow": [
        2,
        {
          "hoist": "all",
          "allow": [
            "resolve",
            "reject",
            "done",
            "next",
            "err",
            "error"
          ]
        }
      ],
      "quotes": [
        2,
        "single",
        {
          "avoidEscape": true,
          "allowTemplateLiterals": true
        }
      ],
      "prettier/prettier": [
        "error",
        {
          "trailingComma": "es5",
          "singleQuote": true,
          "printWidth": 80,
        }
      ],
      "jsx-a11y/href-no-hash": "off",
      "jsx-a11y/anchor-is-valid": [
        "warn",
        {
          "aspects": [
            "invalidHref"
          ]
        }
      ],
      "react-hooks/rules-of-hooks": "error",
      "react-hooks/exhaustive-deps": "warn"
    },
    "plugins": [
      "html",
      "prettier",
      "react-hooks"
    ]
  }`
  
  
The one other comment I'd add is we are using `npm` and not `yarn` on this repo, so we should get rid of the `yarn.lock` file and stick with `package-lock`.

**Comment by Ninja243** [comment]
Eslint config added! Is there anything else that needs changing?

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        "import/prefer-default-export": 0,
+        "import": 0,
+        "func-names": 0,
+        "space-before-function-paren": 0,
+        "comma-dangle": 0,
+        "max-len": 0,
+        "import/extensions": 0,
+        "no-underscore-dangle": 0,
+        "consistent-return": 0,
+        "react/display-name": 1,
```
Comment (aavshr) [inline .eslintrc.js:73]: Can you please remove `react` and `jsx` rules?

File: package.json [hunk 2]
```diff
@@ -9,6 +9,10 @@
   "author": "Deta",
   "license": "MIT",
   "dependencies": {
+    "eslint-config-airbnb": "^18.2.1",
```
Comment (aavshr) [inline package.json:12]: these should only be dev dependencies, also we don't require
`jsx` and `react` plugins.

**Review by aavshr — CHANGES_REQUESTED** [review]
Thanks @Ninja243 , left some comments.

**Review by aavshr — APPROVED** [review]
thanks @mweyaruider , looks good on my end!

**Pull request merged**